ds3-sdk/src/test/java/com/spectralogic/ds3client/helpers/strategy/transferstrategy/TransferStrategy_Test.java

```diff
@@ -128,18 +140,130 @@ private TransferMethod transferMethod(final Set<String> blobsTransferredSuccessf
                                           final Set<String> blobsThatFailed,
                                           final AtomicInteger numTimesTransferCalled)
     {
-        return new TransferMethod() {
-            @Override
-            public void transferJobPart(final JobPart jobPart) throws IOException {
-                numTimesTransferCalled.incrementAndGet();
-
-                if (jobPart.getBlob().getName().equals(BLOB2)) {
-                    blobsThatFailed.add(jobPart.getBlob().getName());
-                    throw new IOException(BLOB2);
-                } else {
-                    blobsTransferredSuccessfully.add(jobPart.getBlob().getName());
-                }
+        return jobPart -> {
+            numTimesTransferCalled.incrementAndGet();
+
+            if (jobPart.getBlob().getName().equals(BLOB2)) {
+                blobsThatFailed.add(jobPart.getBlob().getName());
+                throw new IOException(BLOB2);
+            } else {
+                blobsTransferredSuccessfully.add(jobPart.getBlob().getName());
             }
         };
     }
+
+    @Test
+    public void testCanceledTransferDoesNotTryToGetBlobs() throws IOException {
+        final Ds3Client ds3Client = Mockito.mock(Ds3Client.class);
+
+        final CanceledBlobStrategy blobStrategy = new CanceledBlobStrategy();
+
+        final TransferStrategy transferStrategy = new TransferStrategyBuilder()
+                .withMasterObjectList(masterObjectList())
+                .withBlobStrategy(blobStrategy)
+                .withChannelBuilder(new AChannelBuilder())
+                .withDs3Client(ds3Client)
+                .makePutTransferStrategy();
+
+        transferStrategy.transfer();
+
+        assertEquals(4, blobStrategy.numAttemptsToUse());
+
+        blobStrategy.reset();
+
+        new Thread(() -> {
+            try {
+                transferStrategy.cancel();
+            } catch (final Throwable t) {
+                System.out.println();
```

Comment:
is this line meaning to print the throwable?

Reply:
I'm with Rachel on this, it seems like we should be expecting a specific exception to be thrown here.

Reply:
I'll fix that.